lib/badge-data.js

```diff
@@ -69,7 +69,8 @@ function makeLogo(defaultNamedLogo, overrides) {
 
   if (isDataUri(maybeDataUri)) {
     // +'s are replaced with spaces when used in query params, this returns them to +'s - #1527
-    return maybeDataUri.replace(/\s/g, '+');
+    // remove all whitespace characters (newlines tabs etc) - #1540
+    return maybeDataUri.replace(/ /g, '+').replace(/\s/g, '');
```

Comment:
I think the behavior here is good, which is to be lenient about whitespace within something that's supposed to be base64-encoded. It makes it more forgiving to people's copy and pasting.

However the semantics could be clearer. Could I suggest this change?

```js
const maybeDataUri = prependPrefix(logo, 'data:').replace(/ /g, '+').replace(/\s/g, '');
```

In other words, apply the leniency transform, _then_ pass the transformed value to `isDataUri`.

Reply:
Thanks!,
Didn't even think of doing this.
:+1: 

I think i will either have to add a check for `logo === undefined`
```javascript
const maybeDataUri = logo === undefined ? undefined : prependPrefix(logo, 'data:').replace(/ /g, '+').replace(/\s/g, '');
```
or change the `prependPrefix` function
```diff
- function prependPrefix(s, prefix) {
-   if (s === undefined) {
-     return undefined;
-   }
- 
+ function prependPrefix(s = '', prefix) {
    s = '' + s;

    if (s.startsWith(prefix)) {
      return s;
    } else {
      return prefix + s;
    }
  }
```

Reply:
Our makeLogo function has three checks for undefined. We could check it once and return appropriately. That way we don’t need to keep re-handling it along the way.

Reply:
Thanks!